### ISSUE TITLE
liquid democracy subnode ranking

### ISSUE THREAD

**Issue by codegod100** [flancian/agora-server#59]
I had an idea for liquid democracy for ranking. what if we made subnodes moveable and you could change the order and then you select a ranking delegate and subnodes are ranked according to how your delegate ranked them. You can rank your own subnodes for a node and they will override your delegates

**Comment by codegod100** [comment]
and then someone can delegate you and their ranking will be an amalagation of your own rankings and your delegates rankings

**Comment by codegod100** [comment]
we could enable this for cross agora ranking across the poleis and essentially have a ranking web of trust

**Comment by flancian** [comment]
Yes, I agree, this would be cool! Some quick thoughts on how it could work:

1. We could store votes in user controlled repositories, with the same approach we're taking for implementing writes from social media (which I'm working on now-ish). This could be as simple as a stream of events like a log, "[[flancian]] upvoted [[@user/subnode]] on <date>", "[[flancian]] delegated voting to [[vera]] on <date>".
2. We could put this on (optional :)) sqlite.
3. We could use an independent database somehow, say a distributed ledger, controlled completely client side?

**Comment by codegod100** [comment]
what if we used IPFS to store agora gardens? then people can store whatever they want in their garden and we can cryptographically verify it's them. We don't need to make this mandatory but we could be like "want to enable identity based features of the agora? publish on IPFS" this essentially solves our identity issue for cross-agora stuff

**Comment by codegod100** [comment]
IPNS (the ipfs name service) lets you bind ipns names to domain names so we could do domain based identity like user.anagora.org maybe have a separate domain for agora users so it looks nicer.
